dashboards-observability/public/components/explorer/data_grid.tsx

```diff
@@ -22,20 +23,49 @@ export function DataGrid(props: DataGridProps) {
     rowsAll,
     explorerFields
   } = props;
+  const [limit, setLimit] = useState(PAGE_SIZE);
+  const loader = useRef<HTMLDivElement>(null);
+
+  const handleObserver = useCallback((entries) => {
+    if (entries[0].isIntersecting) {
+      setLimit((limit) => limit + PAGE_SIZE);
+    }
+  }, []);
+
+  useEffect(() => {
+    if (!loader.current || limit >= rows.length + PAGE_SIZE) return;
+
+    const observer = new IntersectionObserver(handleObserver, {
+      root: null,
+      rootMargin: '500px',
+      threshold: 0,
+    });
+    observer.observe(loader.current);
+
+    return () => observer.disconnect();
```

Comment:
Question: Is this just the same thing as 
```
observer.disconnect();
return;
```

Reply:
no, the returned function in useEffect is used for clean up, which will be called when component unmounts. with `observer.disconnect(); return;` then observer is disconnected immediately and won't have any effect